keps/sig-storage/20200212-volume-group.md

```diff
@@ -0,0 +1,1067 @@
+  // in the subsequent `ListVolumeGroups` call. This field is OPTIONAL. If
+  // not specified (zero value), it means there is no restriction on the
+  // number of entries that can be returned.
+  // The value of this field MUST NOT be negative.
+  int32 max_entries = 1;
+
+  // A token to specify where to start paginating. Set this field to
+  // `next_token` returned by a previous `ListVolumeGroups` call to get the
+  // next page of entries. This field is OPTIONAL.
+  // An empty string is equal to an unspecified field value.
+  string starting_token = 2;
+}
+
+message ListVolumeGroupsResponse {
+  option (alpha_message) = true;
+
+  message Entry {
+    // This field is REQUIRED
+    VolumeGroup volume_group = 1;
+  }
+
+  repeated Entry entries = 1;
+
+  // This token allows you to get the next page of entries for
+  // `ListVolumeGroups` request. If the number of entries is larger than
+  // `max_entries`, use the `next_token` as a value for the
+  // `starting_token` field in the next `ListVolumeGroups` request. This
+  // field is OPTIONAL.
+  // An empty string is equal to an unspecified field value.
+  string next_token = 2;
+}
+```
+
+#### CreateVolumeGroupSnapshot
+
+The purpose of this call is to request the creation of a multi-volume snapshot. Group snapshots can be created from existing volume group. Note that calls to this function must be idempotent - the function may be called multiple times for the same name - the group snapshot must only be created once.
+
+```
+message CreateVolumeGroupSnapshotRequest {
+  option (alpha_message) = true;
+
+  // suggested name for a group snapshot (required for idempotent)
+  // This field is REQUIRED.
+  string name = 1;
+
+  // identifier indicates which volume group is used to take
+  // group snapshot
+  // This field is REQUIRED.
+  string source_volume_group_id = 2;
+
+  // volume ids of the volumes in the source group. This field is REQUIRED.
+  // This is needed because some storage systems does not have a group persisted
+  // on the storage system until the time to take a group snapshot
+  repeated string volume_ids = 3;
+
+  // secrets required for snapshot creation (pulled from VolumeSnapshotClass)
+  // This field is OPTIONAL.
+  map<string, string> secrets = 4 [(.csi.v1.csi_secret) = true];
+
+  // params passed from VolumeSnapshotClass
+  // This field is OPTIONAL.
+  map<string, string> parameters = 5;
+}
+
+message CreateVolumeGroupSnapshotResponse {
+  option (alpha_message) = true;
+
+  // Contains all attributes of the newly created group snapshot.
+  // This field is REQUIRED.
+  VolumeGroupSnapshot group_snapshot = 1;
+}
+
+message VolumeGroupSnapshot {
+  option (alpha_message) = true;
+
+  // The identifier for this group snapshot, generated by the plugin.
+  // This field is REQUIRED.
+  string group_snapshot_id = 1;
+
+  // A list of snapshots created. Snapshot is the same
+  // definition as Snapshot definition used in CSI.
+  // This field is REQUIRED.
```

Comment:
Change this to `Optional` as some storage systems cannot return individual snapshots.